puppet/zulip/manifests/golang.pp

```diff
@@ -19,4 +19,17 @@
     target  => $dir,
     require => Zulip::Sha256_tarball_to['golang'],
   }
+
+  file { $dir:
+    ensure  => directory,
+    require => Zulip::Sha256_tarball_to['golang'],
+  }
+
+  tidy { '/srv/golang-*':
+    path    => '/srv/',
+    recurse => 1,
+    rmdirs  => true,
+    matches => 'golang-*',
+    require => File[$dir],
```

Comment:
Hmm, I'm not sure I feel great about having code that removes anything with a non-zulip-mentioning prefix in /srv/ automatically -- seems like it could end up deleting something that we didn't create.

Reply:
(Possibly the right fix is to rename to `/srv/zulip-golang`)

Reply:
I don't feel all that bad about it.  Renaming to `/srv/zulip-golang` is a bit ugly because we still need to clean up the versions that were installed before we did that rename.

Reply:
Yeah but I think those are basically only going to be on our own Zulip-hosted servers, right?  Aside from the newer go-based stuff, we had a pretty clear convention that things in /srv/ had a `zulip-` prefix; since we're moving to having Go stuff there by default, it seems like we should try to restore that convention while it's still doable.

(If we leak a `/srv/golang-` directory one time, I think we can handle that via "Upgrade notes")

Reply:
Why *do* we put these all in a shared namespace, anyway? I’m sure it made sense at the time when there were one or two of these, but now that we have

* /srv/golang
* /srv/golang-*
* /srv/smokescreen-src-*
* /srv/zulip
* /srv/zulip-aws-tools
* /srv/zulip-emoji-cache
* /srv/zulip-node
* /srv/zulip-npm-cache
* /srv/zulip-puppet-cache
* /srv/zulip-py3-venv
* /srv/zulip-venv-cache
* /srv/zulip-yarn

it seems clear to me that something like /var/lib/zulip/XXX would make more sense as a template than /srv/zulip-XXX.

Reply:
I was going to post the same thought, but that also feels like a big migration that probably doesn't belong in this PR (which we plan to backport to the 4.x series).  

I'm not sure it makes sense to invest in that sort of larger reorganization instead of more systematic changes like a `nix` migration, but it's possible it does.  Would be a good chat.zulip.org thread; if we're going to do such a reorganization, we probably want to do it when the 5.0 release is still a few weeks in the future.

Reply:
I'm generally in favor of this idea, but I'm with Tim that I'd rather not shoehorn it into this PR.

I've done the rename to /srv/zulip-XXX for now, to make that future transition perhaps easier.